pkg/dockerregistry/server/metrics/metrics.go

```diff
@@ -0,0 +1,57 @@
+package metrics
+
+import (
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	Register()
+}
+
+const (
+	registryNamespace = "openshift"
+	registrySubsystem = "registry"
+)
+
+var (
+	RegistryAPIRequests = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Namespace: registryNamespace,
+			Subsystem: registrySubsystem,
+			Name:      "request_duration_microseconds",
+			Help:      "Request latency summary in microseconds for each operation",
+		},
+		[]string{"operation", "name"},
+	)
+)
+
+var registerMetrics sync.Once
+
+// Register all metrics.
+func Register() {
```

Comment:
@smarterclayton Not a problem. I call `Register()` from `pkg/cmd/dockerregistry.Execute()` if metrics enabled.